pkg/core/namespace.go

```diff
@@ -210,6 +217,15 @@ func (kc *kubectlClient) createGcrSecret(options NamespaceInitOptions, labels ma
 	return kc.createBasicAuthSecret(options.NamespaceName, options.SecretName, "_json_key", string(token), "https://gcr.io", labels)
 }
 
+func (kc *kubectlClient) createRegistrySecret(options NamespaceInitOptions, labels map[string]string) error {
+	username := options.RegistryUser
+	password, err := readPassword(fmt.Sprintf("Enter password for user %q", username))
+	if err != nil {
+		return err
+	}
+	return kc.createBasicAuthSecret(options.NamespaceName, options.SecretName, username, password, options.RegistryServer, labels)
```

Comment:
should we add th eprotocol `https://` if it is not provided? Seems all example here use a protocol - https://github.com/knative/docs/blob/master/build/auth.md#guiding-credential-selection

Reply:
or should we reject it maybe? We could parse the URL and reject anything that does not have a http or https protocol?

Reply:
should we change the flag to be `--repository-url` then?

Reply:
`repository-url` or `registry-url`?

Reply:
sorry, meant registry-url, although I would prefer to keep registry-host and just add the protocol